### PR TITLE
Support for Discord forum channels

### LINKED ISSUE #122: [Features] Support for Discord forum channels
**Is your feature request related to a problem? Please describe.**
Feature is not related to a problem.

**Describe the solution you'd like**
In addition to the text channels, Discord also has forum channels in which each discussion is held in threads, allowing conversations to be kept clearly organised.
WME Send-to-slack should support these Forum channels.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by Uwe4Waze**
Fixes #122

Proposed changes in this pull request:
- Discord: supporting Forum channel
- Discord: auto-detecting channel type (Text channel/Forum channel)
- Discord: slack-compatible Webhook URLs remain supported; no database changes needed
- Discord: slightly changed request format
- Discord: added signature to request to identify script

- [x] PR as been tested
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/tunisiano187/Chocolatey-packages/blob/master/.github/CONTRIBUTING.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/tunisiano187/WME-send-to-slack/123)

### CODE REVIEW

File: WME-send-to-slack.user.js [hunk 1]
```diff
@@ -530,67 +545,84 @@ ${closureTelegramDetails}${telegramDetails}`;
...
+                        url: serverDB[localStorage.getItem('WMESTSServer')][key][chanel],
+                        error: function (x, y, z) {
+                            log('Slack error : ' + x + ' ' + y + ' ' + z);
+                        }
+                    });
                     log(TextToSend);
                     sent=sent+1;
                     break;
-                case "gform":
+                    case "discord":
```
Comment (GyllieGyllie) [inline WME-send-to-slack.user.js]: Not sure if this is github but indenting on the cases looks wrong?

File: WME-send-to-slack.user.js [hunk 2]
```diff
@@ -530,67 +545,84 @@ ${closureTelegramDetails}${telegramDetails}`;
...
+                        log('Channel : ' + chanel);
+                        var actionicon = "";
+                        log(iconaction);
+                        switch(iconaction.toLowerCase()) {
```
Comment (GyllieGyllie) [inline WME-send-to-slack.user.js]: Not a fan that this whole switch is duplicated now, maybe extract it into a function

**Review by GyllieGyllie — CHANGES_REQUESTED** [review]

File: WME-send-to-slack.user.js (hunk 1 above)
Comment (Uwe4Waze) [reply WME-send-to-slack.user.js]: forgot to reformat the block; I will catch up

File: WME-send-to-slack.user.js (hunk 2 above)
Comment (Uwe4Waze) [reply WME-send-to-slack.user.js]: the block can be moved in front of the outer switch block _switch (key.toLowerCase()) {_

Comment (Uwe4Waze) [reply WME-send-to-slack.user.js]: done

File: WME-send-to-slack.user.js (hunk 1 above)
Comment (Uwe4Waze) [reply WME-send-to-slack.user.js]: done

Comment (GyllieGyllie) [reply WME-send-to-slack.user.js]: Thanks, will test these as quickly as possible before merging!

**Comment by GyllieGyllie** [comment]
Hey @Uwe4Waze,

I've been looking over this code again and wonder if we need the auto detect for channel type or if we can just add it into the data that holds the webhooks anyways. This way it's less prone to issues with the auto detection logic.

Open to hear what you think.

**Comment by Uwe4Waze** [comment]
Hi @GyllieGyllie,
of course we can add the channel type information to the database structure.
To get this information, you need to ask the community admins if they are still using the script with text channels or if they have switched to forum channels in the meantime and therefore do not (cannot) use the script at the moment. The latter is the case for Germany, for example.
If a community is planning to change the channel type, this must be coordinated in order to provide an adapted WMESTSData.js in a timely manner.
Greetings Uwe

**Comment by GyllieGyllie** [comment]
I will merge it as is for now so it at least works again while we look into the other option

**Pull request merged**